### PR TITLE
Implement GET `/users/username/{username}` endpoint for APIv4

### PR DESCRIPTION
**Pull request by ruzette**
#### Summary

Implement GET `/users/username/{username}` endpoint for APIv4.

#### Checklist

- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (mattermost/mattermost-api-reference#87)
- [x] All new/modified APIs include changes to the drivers
- [x] Changed username test case to have a maximum of 22 characters.

### CODE REVIEW

File: api4/params.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ type ApiParams struct {
 	CommandId string
 	HookId    string
 	EmojiId   string
+	UserName  string
```
Comment (jwilander) [inline api4/params.go]: Can we change `UserName` to `Username`? That's how we spell it in other places

File: api4/context.go [hunk 2]
```diff
@@ -358,3 +358,19 @@ func (c *Context) RequireChannelId() *Context {
 	}
 	return c
 }
+
+func (c *Context) RequireUserName() *Context {
```
Comment (jwilander) [inline api4/context.go]: Here as well, can we make this `RequireUsername()`?

**Comment by enahum** [comment]
@ruzette needs a rebase and we're good to go

**Comment by jwilander** [comment]
@ruzette we merged your other one first so this has conflicts now, can you give it a rebase? Feel free to DM me on pre-release if you need help with the rebase :)

**Pull request merged**